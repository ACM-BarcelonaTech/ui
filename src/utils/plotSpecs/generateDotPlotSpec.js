const generateSpec = (config, plotData) => {
  let legend = [];

  if (config.legend.enabled) {
    legend = [
      {
        title: 'Average expression',
        titleColor: config.colour.masterColour,
        orient: config.legend.position,
        labelFont: config.fontStyle.font,
        titleFont: config.fontStyle.font,
        type: 'gradient',
        fill: 'color',
        direction: 'horizontal',
      },
      {
        title: 'Percent Exp. (%)',
        titleColor: config.colour.masterColour,
        orient: config.legend.position,
        labelFont: config.fontStyle.font,
        titleFont: config.fontStyle.font,
        size: 'dotSize',
        format: 's',
        symbolStrokeColor: '#4682b4',
        symbolStrokeWidth: 2,
        symbolOpacity: 0.5,
        symbolType: 'circle',
      },
    ];
  }

  return {
    $schema: 'https://vega.github.io/schema/vega/v5.json',
    width: config.dimensions.width,
    height: config.dimensions.height,
    autosize: { type: 'fit', resize: true },
    background: config.colour.toggleInvert,
    padding: 5,

    data: [
      {
        name: 'plotData',
        values: plotData,
        transform: [
          {
            type: 'formula',
            as: 'size',
            expr: 'datum.cellsFraction * 100',
          },
        ],
      },
    ],

    scales: [
      {
        name: 'x',
        type: 'point',
        range: 'width',
        domain: { data: 'plotData', field: 'gene' },
        padding: 0.2,
      },
      {
        name: 'y',
        type: 'point',
        range: 'height',
        padding: 0.2,
        domain: {
          data: 'plotData',
          field: 'cluster',
        },
      },
      {
        name: 'dotSize',
        type: 'linear',
        domain: {
          data: 'plotData',
          field: 'size',
        },
        range: [
          0,
          100,
        ],
      },
      {
        name: 'color',
        type: 'linear',
        domain: {
          data: 'plotData',
          field: 'AvgExpression',
        },
        range: {
          scheme: 'blues',
        },
      },
    ],

    axes: [
      {
        orient: 'bottom',
        scale: 'x',
        zindex: 1,
        grid: true,
        title: config.axes.xAxisText,
        titleFont: config.fontStyle.font,
        labelFont: config.fontStyle.font,
        labelColor: config.colour.masterColour,
        tickColor: config.colour.masterColour,
        gridColor: config.colour.masterColour,
        gridOpacity: (config.axes.gridOpacity / 20),
        gridWidth: (config.axes.gridWidth / 20),
        offset: config.axes.offset,
        titleFontSize: config.axes.titleFontSize,
        titleColor: config.colour.masterColour,
        labelFontSize: config.axes.labelFontSize,
        domainWidth: config.axes.domainWidth,
        labelAngle: config.axes.xAxisRotateLabels ? 45 : 0,
        labelAlign: config.axes.xAxisRotateLabels ? 'left' : 'center',
      },
      {
        orient: 'left',
        scale: 'y',
        zindex: 1,
        grid: true,
        gridColor: config.colour.masterColour,
        gridOpacity: (config.axes.gridOpacity / 20),
        gridWidth: (config.axes.gridWidth / 20),
        tickColor: config.colour.masterColour,
        offset: config.axes.offset,
        title: config.axes.yAxisText,
        titleFont: config.fontStyle.font,
        labelFont: config.fontStyle.font,
        labelColor: config.colour.masterColour,
        titleFontSize: config.axes.titleFontSize,
        titleColor: config.colour.masterColour,
        labelFontSize: config.axes.labelFontSize,
        domainWidth: config.axes.domainWidth,
      },
    ],

    marks: [
      {
        type: 'symbol',
        shape: 'circle',
        zindex: 2,
        from: {
          data: 'plotData',
        },
        encode: {
          enter: {
            xc: {
              scale: 'x',
              field: 'gene',
            },
            yc: {
              scale: 'y',
              field: 'cluster',
            },
            size: {
              scale: 'dotSize',
              field: 'size',
            },
            fill: {
              scale: 'color',
              field: 'AvgExpression',
            },
          },
        },
      },
    ],
    legends: legend,
    title:
    {
      text: config.title.text,
      color: config.colour.masterColour,
      anchor: config.title.anchor,
      font: config.fontStyle.font,
      dx: 10,
      fontSize: config.title.fontSize,
    },
  };
};

export {
  // eslint-disable-next-line import/prefer-default-export
  generateSpec,
};
