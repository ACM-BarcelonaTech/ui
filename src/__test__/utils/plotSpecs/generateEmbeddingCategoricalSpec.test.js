import _ from 'lodash';

import { filterCells } from 'utils/plotSpecs/generateEmbeddingCategoricalSpec';

import {
  createHierarchyFromTree,
  createPropertiesFromTree,
} from 'redux/reducers/cellSets/helpers';

const { cellSets: mockCellSets } = require('__test__/data/cell_sets.json');

describe('generateEmbeddingCategoricalSpec', () => {
  let mockCellSetsReduxObject;

  beforeEach(() => {
    mockCellSetsReduxObject = {
      properties: createPropertiesFromTree(mockCellSets),
      hierarchy: createHierarchyFromTree(mockCellSets),
    };
  });

  it('filterCells generates correct data with all louvain clusters', () => {
    const result = filterCells(mockCellSetsReduxObject, 'All', 'louvain');

    const expectedCellSetKeys = mockCellSets
      .find(({ key }) => key === 'louvain').children
      .map(({ key, name, color }) => ({ key, name, color }));

    // Contains all the louvain cellSets (and in the correct order)
    expect(result.cellSetLegendsData).toEqual(expectedCellSetKeys);

    // CellIds are filtered fine
    expect(result).toMatchSnapshot();
  });

  it('filterCells generates correct louvain clusters data with one sample in particular', () => {
    const result = filterCells(mockCellSetsReduxObject, '98c7a0ed-d086-4df8-bf94-a9ee6edb793f', 'louvain');

    const louvainClustersWithinSample = ['louvain-0', 'louvain-1', 'louvain-2'];

    const expectedCellSetKeys = mockCellSets
      .find(({ key }) => key === 'louvain').children
      .filter(({ key }) => louvainClustersWithinSample.includes(key))
      .map(({ key, name, color }) => ({ key, name, color }));

    // Contains the 3 expected louvain cellSets
    expect(result.cellSetLegendsData).toEqual(expectedCellSetKeys);

    // CellIds are filtered fine
    expect(result).toMatchSnapshot();
  });

  it('filterCells generates correct samples data with one sample in particular', () => {
    const sampleKey = 'b62028a1-ffa0-4f10-823d-93c9ddb88898';

    const result = filterCells(mockCellSetsReduxObject, sampleKey, 'sample');

    // Contains one single legend
    expect(result.cellSetLegendsData).toEqual([{
      color: '#8c564b',
      key: sampleKey,
      name: 'KO',
    },
    ]);

    // CellIds are filtered fine
    expect(result).toMatchSnapshot();
  });

  it('works correctly with some repeated names in louvain clusters', () => {
    const mockCellSetsWithRepeatedClusterNames = _.cloneDeep(mockCellSets);

    const louvainCellSets = mockCellSetsWithRepeatedClusterNames.find(({ key }) => key === 'louvain').children;

    louvainCellSets[0].name = 'repeatedName';
    louvainCellSets[2].name = 'repeatedName';
    louvainCellSets[6].name = 'repeatedName';

    mockCellSetsReduxObject = {
      properties: createPropertiesFromTree(mockCellSetsWithRepeatedClusterNames),
      hierarchy: createHierarchyFromTree(mockCellSetsWithRepeatedClusterNames),
    };

    const result = filterCells(mockCellSetsReduxObject, 'All', 'louvain');

    const expectedCellSetKeys = mockCellSetsWithRepeatedClusterNames
      .find(({ key }) => key === 'louvain').children
      .map(({ key, name, color }) => ({ key, name, color }));

    // Contains all the louvain cellSets (and in the correct order)
    expect(result.cellSetLegendsData).toEqual(expectedCellSetKeys);

    // CellIds are filtered fine
    expect(result).toMatchSnapshot();
  });
});
