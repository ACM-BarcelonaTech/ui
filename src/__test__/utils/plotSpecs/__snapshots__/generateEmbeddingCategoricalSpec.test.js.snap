// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterCells generates correct data with all louvain clusters 1`] = `
Object {
  "cellSetLegendsData": Array [
    Object {
      "color": "#e377c2",
      "key": "louvain-0",
      "name": "Cluster 0",
    },
    Object {
      "color": "#8c564b",
      "key": "louvain-1",
      "name": "Cluster 1",
    },
    Object {
      "color": "#d62728",
      "key": "louvain-2",
      "name": "Cluster 2",
    },
    Object {
      "color": "#2ca02c",
      "key": "louvain-3",
      "name": "Cluster 3",
    },
    Object {
      "color": "#ff7f0e",
      "key": "louvain-4",
      "name": "Cluster 4",
    },
    Object {
      "color": "#1f77b4",
      "key": "louvain-5",
      "name": "Cluster 5",
    },
    Object {
      "color": "#f8e71c",
      "key": "louvain-6",
      "name": "Cluster 6",
    },
    Object {
      "color": "#3957ff",
      "key": "louvain-7",
      "name": "Cluster 7",
    },
    Object {
      "color": "#d3fe14",
      "key": "louvain-8",
      "name": "Cluster 8",
    },
    Object {
      "color": "#c9080a",
      "key": "louvain-9",
      "name": "Cluster 9",
    },
    Object {
      "color": "#fec7f8",
      "key": "louvain-10",
      "name": "Cluster 10",
    },
    Object {
      "color": "#0b7b3e",
      "key": "louvain-11",
      "name": "Cluster 11",
    },
    Object {
      "color": "#0bf0e9",
      "key": "louvain-12",
      "name": "Cluster 12",
    },
    Object {
      "color": "#c203c8",
      "key": "louvain-13",
      "name": "Cluster 13",
    },
  ],
  "filteredCells": Array [
    Object {
      "cellId": 1,
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
    },
    Object {
      "cellId": 2,
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
    },
    Object {
      "cellId": 3,
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
    },
    Object {
      "cellId": 4,
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
    },
    Object {
      "cellId": 5,
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
    },
    Object {
      "cellId": 6,
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
    },
    Object {
      "cellId": 7,
      "cellSetKey": "louvain-1",
      "cellSetName": "Cluster 1",
      "color": "#8c564b",
    },
    Object {
      "cellId": 8,
      "cellSetKey": "louvain-1",
      "cellSetName": "Cluster 1",
      "color": "#8c564b",
    },
    Object {
      "cellId": 9,
      "cellSetKey": "louvain-1",
      "cellSetName": "Cluster 1",
      "color": "#8c564b",
    },
    Object {
      "cellId": 10,
      "cellSetKey": "louvain-2",
      "cellSetName": "Cluster 2",
      "color": "#d62728",
    },
    Object {
      "cellId": 11,
      "cellSetKey": "louvain-2",
      "cellSetName": "Cluster 2",
      "color": "#d62728",
    },
    Object {
      "cellId": 12,
      "cellSetKey": "louvain-3",
      "cellSetName": "Cluster 3",
      "color": "#2ca02c",
    },
    Object {
      "cellId": 13,
      "cellSetKey": "louvain-3",
      "cellSetName": "Cluster 3",
      "color": "#2ca02c",
    },
    Object {
      "cellId": 14,
      "cellSetKey": "louvain-4",
      "cellSetName": "Cluster 4",
      "color": "#ff7f0e",
    },
    Object {
      "cellId": 15,
      "cellSetKey": "louvain-4",
      "cellSetName": "Cluster 4",
      "color": "#ff7f0e",
    },
    Object {
      "cellId": 16,
      "cellSetKey": "louvain-5",
      "cellSetName": "Cluster 5",
      "color": "#1f77b4",
    },
    Object {
      "cellId": 17,
      "cellSetKey": "louvain-5",
      "cellSetName": "Cluster 5",
      "color": "#1f77b4",
    },
    Object {
      "cellId": 18,
      "cellSetKey": "louvain-6",
      "cellSetName": "Cluster 6",
      "color": "#f8e71c",
    },
    Object {
      "cellId": 19,
      "cellSetKey": "louvain-6",
      "cellSetName": "Cluster 6",
      "color": "#f8e71c",
    },
    Object {
      "cellId": 20,
      "cellSetKey": "louvain-7",
      "cellSetName": "Cluster 7",
      "color": "#3957ff",
    },
    Object {
      "cellId": 21,
      "cellSetKey": "louvain-7",
      "cellSetName": "Cluster 7",
      "color": "#3957ff",
    },
    Object {
      "cellId": 22,
      "cellSetKey": "louvain-8",
      "cellSetName": "Cluster 8",
      "color": "#d3fe14",
    },
    Object {
      "cellId": 23,
      "cellSetKey": "louvain-8",
      "cellSetName": "Cluster 8",
      "color": "#d3fe14",
    },
    Object {
      "cellId": 24,
      "cellSetKey": "louvain-9",
      "cellSetName": "Cluster 9",
      "color": "#c9080a",
    },
    Object {
      "cellId": 25,
      "cellSetKey": "louvain-9",
      "cellSetName": "Cluster 9",
      "color": "#c9080a",
    },
    Object {
      "cellId": 26,
      "cellSetKey": "louvain-10",
      "cellSetName": "Cluster 10",
      "color": "#fec7f8",
    },
    Object {
      "cellId": 27,
      "cellSetKey": "louvain-10",
      "cellSetName": "Cluster 10",
      "color": "#fec7f8",
    },
    Object {
      "cellId": 28,
      "cellSetKey": "louvain-11",
      "cellSetName": "Cluster 11",
      "color": "#0b7b3e",
    },
    Object {
      "cellId": 29,
      "cellSetKey": "louvain-12",
      "cellSetName": "Cluster 12",
      "color": "#0bf0e9",
    },
    Object {
      "cellId": 30,
      "cellSetKey": "louvain-13",
      "cellSetName": "Cluster 13",
      "color": "#c203c8",
    },
  ],
}
`;

exports[`filterCells generates correct louvain clusters data with one sample in particular 1`] = `
Object {
  "cellSetLegendsData": Array [
    Object {
      "color": "#e377c2",
      "key": "louvain-0",
      "name": "Cluster 0",
    },
    Object {
      "color": "#8c564b",
      "key": "louvain-1",
      "name": "Cluster 1",
    },
    Object {
      "color": "#d62728",
      "key": "louvain-2",
      "name": "Cluster 2",
    },
  ],
  "filteredCells": Array [
    Object {
      "cellId": 1,
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
    },
    Object {
      "cellId": 2,
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
    },
    Object {
      "cellId": 3,
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
    },
    Object {
      "cellId": 4,
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
    },
    Object {
      "cellId": 5,
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
    },
    Object {
      "cellId": 6,
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
    },
    Object {
      "cellId": 7,
      "cellSetKey": "louvain-1",
      "cellSetName": "Cluster 1",
      "color": "#8c564b",
    },
    Object {
      "cellId": 8,
      "cellSetKey": "louvain-1",
      "cellSetName": "Cluster 1",
      "color": "#8c564b",
    },
    Object {
      "cellId": 9,
      "cellSetKey": "louvain-1",
      "cellSetName": "Cluster 1",
      "color": "#8c564b",
    },
    Object {
      "cellId": 10,
      "cellSetKey": "louvain-2",
      "cellSetName": "Cluster 2",
      "color": "#d62728",
    },
  ],
}
`;

exports[`filterCells generates correct samples data with one sample in particular 1`] = `
Object {
  "cellSetLegendsData": Array [
    Object {
      "color": "#8c564b",
      "key": "b62028a1-ffa0-4f10-823d-93c9ddb88898",
      "name": "KO",
    },
  ],
  "filteredCells": Array [
    Object {
      "cellId": 11,
      "cellSetKey": "b62028a1-ffa0-4f10-823d-93c9ddb88898",
      "cellSetName": "KO",
      "color": "#8c564b",
    },
    Object {
      "cellId": 12,
      "cellSetKey": "b62028a1-ffa0-4f10-823d-93c9ddb88898",
      "cellSetName": "KO",
      "color": "#8c564b",
    },
    Object {
      "cellId": 13,
      "cellSetKey": "b62028a1-ffa0-4f10-823d-93c9ddb88898",
      "cellSetName": "KO",
      "color": "#8c564b",
    },
    Object {
      "cellId": 14,
      "cellSetKey": "b62028a1-ffa0-4f10-823d-93c9ddb88898",
      "cellSetName": "KO",
      "color": "#8c564b",
    },
    Object {
      "cellId": 15,
      "cellSetKey": "b62028a1-ffa0-4f10-823d-93c9ddb88898",
      "cellSetName": "KO",
      "color": "#8c564b",
    },
    Object {
      "cellId": 16,
      "cellSetKey": "b62028a1-ffa0-4f10-823d-93c9ddb88898",
      "cellSetName": "KO",
      "color": "#8c564b",
    },
    Object {
      "cellId": 17,
      "cellSetKey": "b62028a1-ffa0-4f10-823d-93c9ddb88898",
      "cellSetName": "KO",
      "color": "#8c564b",
    },
    Object {
      "cellId": 18,
      "cellSetKey": "b62028a1-ffa0-4f10-823d-93c9ddb88898",
      "cellSetName": "KO",
      "color": "#8c564b",
    },
    Object {
      "cellId": 19,
      "cellSetKey": "b62028a1-ffa0-4f10-823d-93c9ddb88898",
      "cellSetName": "KO",
      "color": "#8c564b",
    },
    Object {
      "cellId": 20,
      "cellSetKey": "b62028a1-ffa0-4f10-823d-93c9ddb88898",
      "cellSetName": "KO",
      "color": "#8c564b",
    },
  ],
}
`;

exports[`filterCells works correctly with some repeated names in louvain clusters 1`] = `
Object {
  "cellSetLegendsData": Array [
    Object {
      "color": "#e377c2",
      "key": "louvain-0",
      "name": "repeatedName",
    },
    Object {
      "color": "#8c564b",
      "key": "louvain-1",
      "name": "Cluster 1",
    },
    Object {
      "color": "#d62728",
      "key": "louvain-2",
      "name": "repeatedName",
    },
    Object {
      "color": "#2ca02c",
      "key": "louvain-3",
      "name": "Cluster 3",
    },
    Object {
      "color": "#ff7f0e",
      "key": "louvain-4",
      "name": "Cluster 4",
    },
    Object {
      "color": "#1f77b4",
      "key": "louvain-5",
      "name": "Cluster 5",
    },
    Object {
      "color": "#f8e71c",
      "key": "louvain-6",
      "name": "repeatedName",
    },
    Object {
      "color": "#3957ff",
      "key": "louvain-7",
      "name": "Cluster 7",
    },
    Object {
      "color": "#d3fe14",
      "key": "louvain-8",
      "name": "Cluster 8",
    },
    Object {
      "color": "#c9080a",
      "key": "louvain-9",
      "name": "Cluster 9",
    },
    Object {
      "color": "#fec7f8",
      "key": "louvain-10",
      "name": "Cluster 10",
    },
    Object {
      "color": "#0b7b3e",
      "key": "louvain-11",
      "name": "Cluster 11",
    },
    Object {
      "color": "#0bf0e9",
      "key": "louvain-12",
      "name": "Cluster 12",
    },
    Object {
      "color": "#c203c8",
      "key": "louvain-13",
      "name": "Cluster 13",
    },
  ],
  "filteredCells": Array [
    Object {
      "cellId": 1,
      "cellSetKey": "louvain-0",
      "cellSetName": "repeatedName",
      "color": "#e377c2",
    },
    Object {
      "cellId": 2,
      "cellSetKey": "louvain-0",
      "cellSetName": "repeatedName",
      "color": "#e377c2",
    },
    Object {
      "cellId": 3,
      "cellSetKey": "louvain-0",
      "cellSetName": "repeatedName",
      "color": "#e377c2",
    },
    Object {
      "cellId": 4,
      "cellSetKey": "louvain-0",
      "cellSetName": "repeatedName",
      "color": "#e377c2",
    },
    Object {
      "cellId": 5,
      "cellSetKey": "louvain-0",
      "cellSetName": "repeatedName",
      "color": "#e377c2",
    },
    Object {
      "cellId": 6,
      "cellSetKey": "louvain-0",
      "cellSetName": "repeatedName",
      "color": "#e377c2",
    },
    Object {
      "cellId": 7,
      "cellSetKey": "louvain-1",
      "cellSetName": "Cluster 1",
      "color": "#8c564b",
    },
    Object {
      "cellId": 8,
      "cellSetKey": "louvain-1",
      "cellSetName": "Cluster 1",
      "color": "#8c564b",
    },
    Object {
      "cellId": 9,
      "cellSetKey": "louvain-1",
      "cellSetName": "Cluster 1",
      "color": "#8c564b",
    },
    Object {
      "cellId": 10,
      "cellSetKey": "louvain-2",
      "cellSetName": "repeatedName",
      "color": "#d62728",
    },
    Object {
      "cellId": 11,
      "cellSetKey": "louvain-2",
      "cellSetName": "repeatedName",
      "color": "#d62728",
    },
    Object {
      "cellId": 12,
      "cellSetKey": "louvain-3",
      "cellSetName": "Cluster 3",
      "color": "#2ca02c",
    },
    Object {
      "cellId": 13,
      "cellSetKey": "louvain-3",
      "cellSetName": "Cluster 3",
      "color": "#2ca02c",
    },
    Object {
      "cellId": 14,
      "cellSetKey": "louvain-4",
      "cellSetName": "Cluster 4",
      "color": "#ff7f0e",
    },
    Object {
      "cellId": 15,
      "cellSetKey": "louvain-4",
      "cellSetName": "Cluster 4",
      "color": "#ff7f0e",
    },
    Object {
      "cellId": 16,
      "cellSetKey": "louvain-5",
      "cellSetName": "Cluster 5",
      "color": "#1f77b4",
    },
    Object {
      "cellId": 17,
      "cellSetKey": "louvain-5",
      "cellSetName": "Cluster 5",
      "color": "#1f77b4",
    },
    Object {
      "cellId": 18,
      "cellSetKey": "louvain-6",
      "cellSetName": "repeatedName",
      "color": "#f8e71c",
    },
    Object {
      "cellId": 19,
      "cellSetKey": "louvain-6",
      "cellSetName": "repeatedName",
      "color": "#f8e71c",
    },
    Object {
      "cellId": 20,
      "cellSetKey": "louvain-7",
      "cellSetName": "Cluster 7",
      "color": "#3957ff",
    },
    Object {
      "cellId": 21,
      "cellSetKey": "louvain-7",
      "cellSetName": "Cluster 7",
      "color": "#3957ff",
    },
    Object {
      "cellId": 22,
      "cellSetKey": "louvain-8",
      "cellSetName": "Cluster 8",
      "color": "#d3fe14",
    },
    Object {
      "cellId": 23,
      "cellSetKey": "louvain-8",
      "cellSetName": "Cluster 8",
      "color": "#d3fe14",
    },
    Object {
      "cellId": 24,
      "cellSetKey": "louvain-9",
      "cellSetName": "Cluster 9",
      "color": "#c9080a",
    },
    Object {
      "cellId": 25,
      "cellSetKey": "louvain-9",
      "cellSetName": "Cluster 9",
      "color": "#c9080a",
    },
    Object {
      "cellId": 26,
      "cellSetKey": "louvain-10",
      "cellSetName": "Cluster 10",
      "color": "#fec7f8",
    },
    Object {
      "cellId": 27,
      "cellSetKey": "louvain-10",
      "cellSetName": "Cluster 10",
      "color": "#fec7f8",
    },
    Object {
      "cellId": 28,
      "cellSetKey": "louvain-11",
      "cellSetName": "Cluster 11",
      "color": "#0b7b3e",
    },
    Object {
      "cellId": 29,
      "cellSetKey": "louvain-12",
      "cellSetName": "Cluster 12",
      "color": "#0bf0e9",
    },
    Object {
      "cellId": 30,
      "cellSetKey": "louvain-13",
      "cellSetName": "Cluster 13",
      "color": "#c203c8",
    },
  ],
}
`;

exports[`generateData generates correct data with 1 cluster 1`] = `
Object {
  "cellSetLegendsData": Array [
    Object {
      "color": "#1f77b4",
      "key": "louvain-5",
      "name": "Cluster 5",
    },
  ],
  "plotData": Array [
    Object {
      "cellSetKey": "louvain-5",
      "cellSetName": "Cluster 5",
      "color": "#1f77b4",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-5",
      "cellSetName": "Cluster 5",
      "color": "#1f77b4",
      "x": 1,
      "y": 2,
    },
  ],
}
`;

exports[`generateData generates correct data with all louvain clusters 1`] = `
Object {
  "cellSetLegendsData": Array [
    Object {
      "color": "#e377c2",
      "key": "louvain-0",
      "name": "Cluster 0",
    },
    Object {
      "color": "#8c564b",
      "key": "louvain-1",
      "name": "Cluster 1",
    },
    Object {
      "color": "#d62728",
      "key": "louvain-2",
      "name": "Cluster 2",
    },
    Object {
      "color": "#2ca02c",
      "key": "louvain-3",
      "name": "Cluster 3",
    },
    Object {
      "color": "#ff7f0e",
      "key": "louvain-4",
      "name": "Cluster 4",
    },
    Object {
      "color": "#1f77b4",
      "key": "louvain-5",
      "name": "Cluster 5",
    },
    Object {
      "color": "#f8e71c",
      "key": "louvain-6",
      "name": "Cluster 6",
    },
    Object {
      "color": "#3957ff",
      "key": "louvain-7",
      "name": "Cluster 7",
    },
    Object {
      "color": "#d3fe14",
      "key": "louvain-8",
      "name": "Cluster 8",
    },
    Object {
      "color": "#c9080a",
      "key": "louvain-9",
      "name": "Cluster 9",
    },
    Object {
      "color": "#fec7f8",
      "key": "louvain-10",
      "name": "Cluster 10",
    },
    Object {
      "color": "#0b7b3e",
      "key": "louvain-11",
      "name": "Cluster 11",
    },
    Object {
      "color": "#0bf0e9",
      "key": "louvain-12",
      "name": "Cluster 12",
    },
    Object {
      "color": "#c203c8",
      "key": "louvain-13",
      "name": "Cluster 13",
    },
  ],
  "plotData": Array [
    Object {
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-0",
      "cellSetName": "Cluster 0",
      "color": "#e377c2",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-1",
      "cellSetName": "Cluster 1",
      "color": "#8c564b",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-1",
      "cellSetName": "Cluster 1",
      "color": "#8c564b",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-1",
      "cellSetName": "Cluster 1",
      "color": "#8c564b",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-2",
      "cellSetName": "Cluster 2",
      "color": "#d62728",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-2",
      "cellSetName": "Cluster 2",
      "color": "#d62728",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-3",
      "cellSetName": "Cluster 3",
      "color": "#2ca02c",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-3",
      "cellSetName": "Cluster 3",
      "color": "#2ca02c",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-4",
      "cellSetName": "Cluster 4",
      "color": "#ff7f0e",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-4",
      "cellSetName": "Cluster 4",
      "color": "#ff7f0e",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-5",
      "cellSetName": "Cluster 5",
      "color": "#1f77b4",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-5",
      "cellSetName": "Cluster 5",
      "color": "#1f77b4",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-6",
      "cellSetName": "Cluster 6",
      "color": "#f8e71c",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-6",
      "cellSetName": "Cluster 6",
      "color": "#f8e71c",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-7",
      "cellSetName": "Cluster 7",
      "color": "#3957ff",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-7",
      "cellSetName": "Cluster 7",
      "color": "#3957ff",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-8",
      "cellSetName": "Cluster 8",
      "color": "#d3fe14",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-8",
      "cellSetName": "Cluster 8",
      "color": "#d3fe14",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-9",
      "cellSetName": "Cluster 9",
      "color": "#c9080a",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-9",
      "cellSetName": "Cluster 9",
      "color": "#c9080a",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-10",
      "cellSetName": "Cluster 10",
      "color": "#fec7f8",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-10",
      "cellSetName": "Cluster 10",
      "color": "#fec7f8",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-11",
      "cellSetName": "Cluster 11",
      "color": "#0b7b3e",
      "x": 1,
      "y": 2,
    },
    Object {
      "cellSetKey": "louvain-12",
      "cellSetName": "Cluster 12",
      "color": "#0bf0e9",
      "x": 1,
      "y": 2,
    },
  ],
}
`;
